### PR TITLE
Use `gographics/imagick` for webp emote conversion

### PR DESCRIPTION
**Pull request by AnatoleAM**
Converting the raw shell command exec to [gographics/imagick](https://github.com/gographics/imagick)

### CODE REVIEW

File: src/server/api/v2/rest/emotes/create-emote.go [hunk 1]
```diff
@@ -77,7 +75,7 @@ func CreateEmoteRoute(router fiber.Router) {
 			ogFilePath := fmt.Sprintf("%v/og", fileDir) // The original file's path in temp
 
 			// Remove temp dir once this function completes
-			defer os.RemoveAll(fileDir)
+			// defer os.RemoveAll(fileDir)
```
Comment (TroyDota) [inline src/server/api/v2/rest/emotes/create-emote.go:78]: Why is this removed?

File: src/mongo/datastructure/emote.go [hunk 2]
```diff
@@ -66,10 +66,10 @@ func (*emoteUtil) AddSizeMetadata(emote *Emote) ([4]int16, [4]int16, error) {
 func (*emoteUtil) GetFilesMeta(fileDir string) [][]string {
 	// Define sizes to be generated
 	return [][]string{
-		{fmt.Sprintf("%s/1x", fileDir), "1x", "96x32", "20"},
-		{fmt.Sprintf("%s/2x", fileDir), "2x", "144x48", "35"},  // Upscale: 1x * 1.5
-		{fmt.Sprintf("%s/3x", fileDir), "3x", "228x76", "40"},  // Upscale: 2x * 1.585
-		{fmt.Sprintf("%s/4x", fileDir), "4x", "384x128", "60"}, // Upscale: 3x * 1.685
+		{fmt.Sprintf("%s/1x", fileDir), "1x", "96x32", "100"},
```
Comment (TroyDota) [inline src/mongo/datastructure/emote.go:69]: What do these numbers represent?

File: src/server/api/v2/rest/emotes/create-emote.go [hunk 3]
```diff
@@ -253,25 +253,37 @@ func CreateEmoteRoute(router fiber.Router) {
...
+					aw.SetIteratorIndex(ind)
+					img := aw.GetImage()
+					img.ResizeImage(uint(width), uint(height), imagick.FILTER_LANCZOS)
+					mw.AddImage(img)
+					img.Destroy()
+				}
+
+				// Done - convert to WEBP
+				mw.ResetIterator()
+				mw.SetImageFormat("webp:lossless=true,auto-filter=true,method=6")
```
Comment (TroyDota) [inline src/server/api/v2/rest/emotes/create-emote.go:283]: What does method=6 and auto filter mean? and why are we doing lossless?

File: src/server/api/v2/rest/emotes/create-emote.go [hunk 4]
```diff
@@ -253,25 +253,37 @@ func CreateEmoteRoute(router fiber.Router) {
...
+
+				// Write to file
+				err = mw.WriteImages(outFile, true)
```
Comment (TroyDota) [inline src/server/api/v2/rest/emotes/create-emote.go:286]: is this writing to a disk?

File: src/server/api/v2/rest/emotes/create-emote.go (hunk 1 above)
Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:78]: Forgot to uncomment, was for debugging 👍

File: src/mongo/datastructure/emote.go (hunk 2 above)
Comment (AnatoleAM) [reply src/mongo/datastructure/emote.go:69]: File path, emote size, emote width/height, quality factor

File: src/server/api/v2/rest/emotes/create-emote.go (hunk 3 above)
Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: Do you think we should be using lossy?

Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: Method 6 is the more resource-intensive one but produces a better overall image and compression

File: src/server/api/v2/rest/emotes/create-emote.go (hunk 4 above)
Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:289]: Yes

File: src/server/api/v2/rest/emotes/create-emote.go (hunk 3 above)
Comment (TroyDota) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: Can you compare doing lossy vs lossless in filesize and quality?

Comment (TroyDota) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: Lossless is significantly larger i think.

Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: After some testing it doesn't look like these parameters have any effect with imagick. There might be some other way to set them. Removing lossless=true and the rest didn't change the out file

Comment (TroyDota) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: hmm, can we then remove them to not cause confusion since they dont do anything?

Comment (AnatoleAM) [reply src/server/api/v2/rest/emotes/create-emote.go:286]: Will do that, imagemagick should be setting it to lossless by default regardless

**Comment by TroyDota** [comment]
👍
